data/core/doc/init.lua

```diff
@@ -124,9 +124,21 @@ function Doc:get_selection(sort)
   return line1, col1, line2, col2, sort
 end
 
+function Doc:get_selection_text(limit)
+  limit = limit or math.huge
+  local result = {}
+  for idx, line1, col1, line2, col2 in self:get_selections() do
+    if idx > limit then break end
+    if line1 ~= line2 or col1 ~= col2 then
```

Comment:
You can use `Doc:has_selection()` to increase readability

Reply:
`has_selection` checks if there's selection, we need to actually get the selection here.

Reply:
sorry. my miss